ffs/scheduler/cistore/cistore.go

```diff
@@ -59,6 +62,32 @@ func (s *Store) Put(ci ffs.CidInfo) error {
 	return nil
 }
 
+// Query returns a list of CidInfo
+func (s *Store) Query(selector func(ffs.CidInfo) bool) ([]ffs.CidInfo, error) {
```

Comment:
Hey @jsign. My goal is to be able to query for all `CidInfo` associated with a single `ffs` instance. I don't think this does it because the `cistore.Store` is owned buy the scheduler, and data for all `ffs` instances is written to it. There isn't anything about the keys scoping it to an `ffs` instance. 

The ffs instance store `istore.Store` does persist a list of cids associated with the ffs instance. I could query that list, then for each cid, make a call to `cistore.Get` in here. That feels really inefficient to me, and may come with performance problems when there are lots of cids. But I'm also not too familiar with the performance characteristics of kv stores like Badger. Am I overthinking this?

Reply:
Yes, the last paragraph is the correct approach. If that becomes a real problem, we have plenty of options.